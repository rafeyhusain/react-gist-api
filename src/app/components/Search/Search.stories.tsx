import React from 'react';

import Search from './Search';

export default {
  title: 'Component/Search',
  component: Search,
};

export const Default = (): JSX.Element => <Search />;
