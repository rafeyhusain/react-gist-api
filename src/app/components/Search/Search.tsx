import React, { useState } from 'react';

type OnSearch = (text: string) => void;

type SearchProps = {
  onSearch: OnSearch;
};

function Search({ onSearch }: SearchProps): JSX.Element {
  const [text, setText] = useState('rafeyhusain');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value;

    setText(value);
  };

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    onSearch(text)
  };

  return <div className="text-l text-left bg-blue-300">
    <input
      className="ml-2 py-3 px-4 rounded"
      value={text}
      onChange={handleChange}
      placeholder=" e.g. rafeyhusain" />&nbsp;
    <button
      className="bg-yellow-500 hover:bg-yellow-700 text-white font-bold py-2 px-4 rounded"
      onClick={handleClick}>Search</button>
  </div >;
}

export default Search;
