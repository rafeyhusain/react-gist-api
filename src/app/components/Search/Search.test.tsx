import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Search from './Search'

describe("Search component tests", () => {
  test("renders without crash", ()=> {
    const div = document.createElement("div")
  
    ReactDOM.render(<Search onSearch={()=>{}}/>, div)
    ReactDOM.unmountComponentAtNode(div)
  });

  test("Snapshot test for Search", ()=> {
    const component = renderer.create(<Search onSearch={() => {}} />).toJSON()

    expect(component).toMatchSnapshot()
  });
})

