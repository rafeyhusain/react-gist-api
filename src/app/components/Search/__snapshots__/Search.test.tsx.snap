// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component tests Snapshot test for Search 1`] = `
<div
  className="text-l text-left bg-blue-300"
>
  <input
    className="ml-2 py-3 px-4 rounded"
    onChange={[Function]}
    placeholder=" e.g. rafeyhusain"
    value="rafeyhusain"
  />
  Â 
  <button
    className="bg-yellow-500 hover:bg-yellow-400 text-white py-2 px-4 rounded"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
